src/libs/API.js

```diff
@@ -52,14 +52,15 @@ function addAuthTokenToParameters(command, parameters) {
 
     if (isAuthTokenRequired(command) && !parameters.authToken) {
         // If we end up here with no authToken it means we are trying to make an API request before we are signed in.
-        // In this case, we should cancel the current request by pausing the queue and clearing the remaining requests.
+        // In this case, we should cancel the current request and clear the queue.
         if (!authToken) {
             redirectToSignIn();
 
             console.debug('A request was made without an authToken', {command, parameters});
-            Network.pauseRequestQueue();
```

Comment:
@marcaaron Could you please take a look here? 
It's your code and I'm a bit afraid to break something, because I don't see the whole picture yet.

This `pause` here still creates problems for sign in.
If user is already on Sign in page, then `redirectToSignIn()` does nothing (see [SignInRedirect](https://github.com/Expensify/Expensify.cash/blob/a6ba3f26ea8631b43dc34b559e86fd33c9cf69fa/src/libs/actions/SignInRedirect.js#L37))

So the user ends up on Sign In page with paused queue, and Continue button still spins forever.

Reply:
To clarify, which command are we calling that causes `redirectToSignIn()` to be called?

Reply:
That's again a case of a request after the expired session, the one in the original issue.

It's fixed now, because I replaced the listener removal, but I'm able to reproduce when I revert b0f89ce

So if some forgotten request is being sent after the sign out, it no longer breaks in `handleExpiredAuthToken` because of my checks for credentials. Instead it ends up here (because there's no auth token).
And because we are still on Sign in page, the redirect isn't happening, page is not refreshed, queue is paused without unpause.

Reply:
Ok I see now. I'm not sure we want to remove this, because there are cases where we do want to pause the queue so the queued command doesn't `fetch()` here:

https://github.com/Expensify/Expensify.cash/blob/b2c03fc5a8bd0d2c5a3f67847d132af1d41702e1/src/libs/Network.js#L59-L63

Throwing would stop the `fetch()` from happening, but might have other unintended effects as we are not catching this error in `processNetworkRequest()`.

Perhaps there's some other way to unpause the queue again?

I'm not too sure why we are preventing a redirect to sign in so maybe it would be reasonable to remove that check and just unpause the network request queue once storage has been cleared.

Reply:
You mean to remove the check for sign in the url?

https://github.com/Expensify/Expensify.cash/blob/a6ba3f26ea8631b43dc34b559e86fd33c9cf69fa/src/libs/actions/SignInRedirect.js#L37-L40

In this case it might break the flow even more mysteriously for the user, because it might redirect and refresh the page.
User is entering email on the Sign in page, forgotten request suddenly fires (unseen by the user) and the page suddenly refreshes

Reply:
Ah yep, that's a great point. We probably don't want to do that.

Reply:
Another idea discussed was to return a `Promise` from `signInRedirect()` and unpause once we resolve it. But this feels slightly hacky since what we really would like to do is let `processRequestQueue()` finish running and then unpause the request queue again.

Reply:
FYI I've updated the credentials check, but still thinking how to resolve `signInRedirect` issue